File: regular_assignments/day_024/print_first_n_natural_numbers/solution.rb
class Solution
  def call
    solve(5)
    puts "\n"
    solve(9)
    puts "\n"
    solve(10)
    puts "\n"
    solve(77)
  end

  def solve(n)
    cprint(1, n)
  end

  def cprint(i, n)
    if i > n
      return
    else
      print i
      print " "
      cprint(i+1, n)
    end
  end
end

File: regular_assignments/day_024/print_first_n_natural_numbers/solution.go
package printFirstNNaturalNumbers

import "fmt"

func Call() {
	solve(10)
}

func solve(n int) {
	print(1, n)
}

func print(i, n int) {
	if i > n {
		return
	} else {
		fmt.Printf("%v ", i)
		print(i+1, n)
	}
}
